### PR TITLE
prov/verbs bugfixes

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
The series includes the next bug fixes and refactoring per commit:

- remove odd verbs_av.c file, all content is moved to rdm/verbs_av_ep_rdm.c file
- add check for shared AV and return error
- fill buf field of cq_entry 
- fix completion flag for send msg

@a-ilango @hoopoepg please review
@shefty

### CODE REVIEW

**Comment by hoopoepg** [comment]
sounds good

File: prov/verbs/src/ep_rdm/verbs_cq_ep_rdm.c [hunk 1]
```diff
@@ -70,6 +70,8 @@ static ssize_t fi_ibv_rdm_tagged_cq_readfrom(struct fid_cq *cq, void *buf,
 		entry[ret].op_context = cq_entry->context;
 		entry[ret].flags = (cq_entry->comp_flags & ~FI_COMPLETION);
 		entry[ret].len = cq_entry->len;
+		entry[ret].buf = (cq_entry->comp_flags & FI_TRANSMIT) ?
+			cq_entry->src_addr : cq_entry->dest_buf;
```
Comment (a-ilango) [inline prov/verbs/src/ep_rdm/verbs_cq_ep_rdm.c]: The man page states the buf field is valid only for multi-recv operations.

Comment (evgeny-leksikov) [inline prov/verbs/src/ep_rdm/verbs_cq_ep_rdm.c]: Yes, and multi-recv is under development. Also, I do not see the statement that the buffer must be invalid for other cases :) Anyway, we'll have if-statement here. So, I would keep it as is for now.

Comment (a-ilango) [inline prov/verbs/src/ep_rdm/verbs_cq_ep_rdm.c]: 👍

**Pull request merged**